File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/SetPriority.py
from enum import Enum

from ravendb import SetIndexesPriorityOperation
from ravendb.documents.indexes.definitions import IndexPriority
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class SetPriority(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_set_priority(self):
        with self.embedded_server.get_document_store("SetPriority") as store:
            # region set_priority_single
            # Define the set priority operation
            # Pass index name & priority
            set_priority_op = SetIndexesPriorityOperation(IndexPriority.HIGH, "Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            # An exception will be thrown if index does not exist
            store.maintenance.send(set_priority_op)
            # endregion

            # region set_priority_multiple
            # Define the set priority operation, pass multiple index names
            set_priority_op = SetIndexesPriorityOperation(IndexPriority.LOW, "Orders/Totals", "Orders/ByCompany")

            # Execute the operation by passing it to maintenance.send
            # An exception will be thrown if any of the specified indexes do not exist
            store.maintenance.send(set_priority_op)
            # endregion

    class Foo:
        # region syntax_1
        class SetIndexesPriorityOperation(VoidMaintenanceOperation):
            def __init__(self, priority: IndexPriority, *index_names: str): ...

        # endregion

    class Priority:
        # region syntax_2
        class IndexPriority(Enum):
            LOW = "Low"
            NORMAL = "Normal"
            HIGH = "High"

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/SetPriority.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexPriority;
use RavenDB\Documents\Operations\Indexes\IndexPriorityParameters;
use RavenDB\Documents\Operations\Indexes\SetIndexesPriorityOperation;

class SetPriority
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region set_priority_single
            // Define the set priority operation
            // Pass index name & priority
            $setPriorityOp = new SetIndexesPriorityOperation("Orders/Totals", IndexPriority::high());

            // Execute the operation by passing it to Maintenance.Send
            // An exception will be thrown if index does not exist
            $store->maintenance()->send($setPriorityOp);
            # endregion
        } finally {
            $store->close();
        }

        $store = new DocumentStore();
        try {
            # region set_priority_multiple
            // Define the index list and the new priority:
            $parameters = new IndexPriorityParameters();
            $parameters->setIndexNames(["Orders/Totals", "Orders/ByCompany"]);
            $parameters->setPriority(IndexPriority::low());

            // Define the set priority operation, pass the parameters
            $setPriorityOp = new SetIndexesPriorityOperation($parameters);

            // Execute the operation by passing it to Maintenance.Send
            // An exception will be thrown if any of the specified indexes do not exist
            $store->maintenance()->send($setPriorityOp);
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax_1
    // Available overloads:
    SetIndexesPriorityOperation(?string $indexName, ?IndexPriority $priority);
    SetIndexesPriorityOperation(?Parameters $parameters);
    # endregion
}
*/

/*
# region syntax_2
class IndexPriority
{
    public static function low(): IndexPriority;
    public static function normal(): IndexPriority;
    public static function high(): IndexPriority;

    public function isLow(): bool;
    public function isNormal(): bool;
    public function isHigh(): bool;
}
# endregion
*/

/*
# region syntax_3
public class IndexPriorityParameters
{
    private ?StringArray $indexNames = null;
    private ?IndexPriority $priority = null;

    // ... getters and setters
}
# endregion
*/
